### ISSUE TITLE
Connect Four: model currently not improving, best parameters for training.

### ISSUE THREAD

**Issue by gemasphi** [facebookincubator/Polygames#13]
Hey,

I'm trying to understand better the convergence speed of zero learning algorithms. The game I am using is Connect Four which should be simple enough, but currently the model isn't getting better. As you can see below it seems to be overfitting to a subpar policy.

Do you know what are the best parameters for training connect four? I used most of the default parameters as I thought that would suffice. 
By the way, currently the hardware I'm using is two Tesla T4s and a CPU with 32 cores. Do you have any idea if the `per_thread_batchsize=64` is ideal?

## Steps to reproduce
  1. `python -m pypolygames traineval --per_thread_batchsize=64  --num_game=32 --model_name=GenericModel --game_name=Connect4 --batchsize=128 --real_time --saving_period=5     --checkpoint_dir="checkpoints" --device cuda:0 cuda:1 --num_rollouts_opponent=1000`

## Observed Results
![2020-03-16-112834_1366x768_scrot](https://user-images.githubusercontent.com/19616447/76755209-63218180-677b-11ea-9d9e-0f2c20d90052.png)

```
loading checkpoint #5...
Playing 100 games of Connect4:
- GenericModel player uses 400 rollouts per actor with 1 actor
- pure MCTS opponent uses 1000 rollouts per actor with 1 actor
@@@eval: win: 20.00, tie: 3.00, loss: 77.00, avg: 21.50
loading checkpoint #10...
@@@eval: win: 20.00, tie: 2.00, loss: 78.00, avg: 21.00
loading checkpoint #15...
@@@eval: win: 20.00, tie: 1.00, loss: 79.00, avg: 20.50
loading checkpoint #20...
@@@eval: win: 24.00, tie: 1.00, loss: 75.00, avg: 24.50
loading checkpoint #25...
@@@eval: win: 21.00, tie: 0.00, loss: 79.00, avg: 21.00
loading checkpoint #30...
@@@eval: win: 9.00, tie: 5.00, loss: 86.00, avg: 11.50
loading checkpoint #35...
@@@eval: win: 18.00, tie: 1.00, loss: 81.00, avg: 18.50
loading checkpoint #40...
@@@eval: win: 20.00, tie: 2.00, loss: 78.00, avg: 21.00
loading checkpoint #45...
@@@eval: win: 25.00, tie: 3.00, loss: 72.00, avg: 26.50
loading checkpoint #50...
@@@eval: win: 30.00, tie: 5.00, loss: 65.00, avg: 32.50
loading checkpoint #55...
@@@eval: win: 23.00, tie: 4.00, loss: 73.00, avg: 25.00
loading checkpoint #60...
@@@eval: win: 18.00, tie: 4.00, loss: 78.00, avg: 20.00
loading checkpoint #65...
@@@eval: win: 21.00, tie: 3.00, loss: 76.00, avg: 22.50
loading checkpoint #70...
@@@eval: win: 21.00, tie: 4.00, loss: 75.00, avg: 23.00
loading checkpoint #75...
@@@eval: win: 17.00, tie: 6.00, loss: 77.00, avg: 20.00
loading checkpoint #80...
@@@eval: win: 18.00, tie: 2.00, loss: 80.00, avg: 19.00
```

**Comment by tscmoo** [comment]
Hi, the loss value is not necessarily a good indicator of convergence as it is self-play. You may just need to train longer, however the default GenericModel as I recall is quite large, so you may want to try with a different model, eg `-model_name ResConvConvLogitModel`
